packages/server-core/src/media/static-resource/static-resource.hooks.ts

```diff
@@ -163,27 +182,26 @@ export default {
       iff(
         isProvider('external'),
         iffElse(
-          checkScope('static_resource', 'read'),
+          (ctx: HookContext) => isAction('admin')(ctx) && checkScope('static_resource', 'read')(ctx),
           [],
-          [verifyScope('editor', 'write'), resolveProjectId(), verifyProjectPermission(['owner', 'editor', 'reviewer'])]
+          [
+            iffElse(
+              hasProjectField,
+              [
+                verifyScope('editor', 'write'),
+                resolveProjectId(),
+                verifyProjectPermission(['owner', 'editor', 'reviewer'])
+              ],
+              [verifyScope('editor', 'write'), resolveProjectsByPermission()]
+            ) as any
+          ]
```

Comment:
Since `verifyScope('editor', 'write')` is checked in both cases, therefore it can be moved before iffElse.

```suggestion
          [
            verifyScope('editor', 'write'),
            iffElse(
              hasProjectField,
              [
                resolveProjectId(),
                verifyProjectPermission(['owner', 'editor', 'reviewer'])
              ],
              [resolveProjectsByPermission()]
            ) as any
          ]
```